pytorch_lightning/overrides/data_parallel.py

```diff
@@ -151,74 +152,98 @@ def parallel_apply(self, replicas, inputs, kwargs):
         return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])
 
 
-class LightningDistributedDataParallel(DistributedDataParallel):
-    """
-    Override the forward call in lightning so it goes to training and validation step respectively
-    """
-    PREPARE_FOR_BACKWARDS = True
-
-    def parallel_apply(self, replicas, inputs, kwargs):
-        return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])
+class LightningDistributedWrapper(torch.nn.Module):
```

Comment:
for later: we should add comments for how this class is to be used

Reply:
done, with example.